hypothesis-python/src/hypothesis/searchstrategy/strategies.py

```diff
@@ -498,14 +613,12 @@ def calc_label(self):
 
     def do_draw(self, data):
         # type: (ConjectureData) -> Ex
-        n = len(self.element_strategies)
-        assert n > 0
-        if n == 1:
-            return data.draw(self.element_strategies[0])
-
-        i = cu.integer_range(data, 0, n - 1)
-
-        return data.draw(self.element_strategies[i], label=self.branch_labels[i])
+        strategy = data.draw(
+            SampledFromStrategy(self.element_strategies).filter(
```

Comment:
The `element_strategies` property currently skips anything which is empty, and that's cached forever.  Either:

1. Keep empty strategies in the list, so they're filtered out here, or
2. Ensure that `some_bundle.is_empty` always returns `False` even when it's currently empty

Reply:
Solution 1 feels like bringing too much of `Bundle` specific code into `OneOfStrategy`, so I choose solution 2.

Besides, it seems logical to always have `Bundle.is_empty == True` given a Bundle strategy raison d'être is to grow over time.
Of course a more advanced way would be to determine if this bundle will grow by looking at the other Rule strategies to see if a non-empty one has this bundle as target. But this seems pretty complex to do so I guess the current solution is enough for the moment ;-)

By the way, this doesn't change the behavior of the system given before that `SearchStrategy.calc_is_empty` was called which always return `False`